renderdoc/core/replay_proxy.cpp

```diff
@@ -1707,6 +1707,21 @@ bool ReplayProxy::SendReplayCommand(ReplayProxyPacket type)
   return true;
 }
 
+// Modify the params that are used when getting the texture data for replay on the current driver.
+void ReplayProxy::ConvertProxyTextureIfNeeded(FetchTexture &tex, GetTextureDataParams &params)
+{
+  if(m_Proxy->IsTextureFormatSupported(tex.format))
+    return;
+
+  tex.format.special = false;
+  tex.format.compCount = 4;
+  tex.format.compByteWidth = 1;
+  tex.format.compType = eCompType_UNorm;
+
+  params.forceRGBA8unorm = true;
+  params.whitePoint = 1.0f;
```

Comment:
This code will only work for a subset of possible formats that aren't supported on a proxy. Not all can just be remapped down to RGBA8_UNORM without loss of information.

It's fine to just special case support for a certain number of formats (e.g. ETC2 and ASTC) that need to be remapped right now and leave others unimplemented, but this code should at least detect those formats specifically and then print an error if trying to remap some other format.

I believe it will also need to be expanded to handle different types of format, e.g. HDR and LDR ASTC, and different different channel counts for ETC/EAC.